### PR TITLE
Pure go sqlite

### PR DESCRIPTION
**Pull request by wizsk**
# Important

Changed cgo sqlite library to pure go library https://github.com/zombiezen/go-sqlite

## Issue ticket number and link

Closes https://github.com/pouriyajamshidi/tcping/issues/152

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [x] If it is a core feature, I have added tests.
- [x] I have run `make check` and there are no failures.

### CODE REVIEW

**Comment by wizsk** [comment]
@pouriyajamshidi let me know if it needs any changes.

**Comment by pouriyajamshidi** [comment]
> @pouriyajamshidi let me know if it needs any changes.

Hi @wizsk,

I am yet to properly review this. Had a quick look yesterday but I was too tired. 

Very busy with work these days.

Excuses for the delay

**Comment by wizsk** [comment]
@pouriyajamshidi No worries man. Review it whenever you get time.

File: db.go [hunk 1]
```diff
@@ -64,26 +65,57 @@ CREATE TABLE %s (
...
+	longest_downtime_end,
+	latency_min,
+	latency_avg,
+	latency_max,
+	start_time,
+	end_time,
+	total_duration) VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?);`
 )
 
 // newDb creates a newDb with the given path and returns `database` struct
```
Comment (pouriyajamshidi) [inline db.go:101]:
```suggestion
// newDb creates a newDb with the given path and returns a pointer to the `database` struct
```

File: db.go [hunk 2]
```diff
@@ -64,26 +65,57 @@ CREATE TABLE %s (
...
+	conn, err := sqlite.OpenConn(dbPath, sqlite.OpenCreate, sqlite.OpenReadWrite)
 	if err != nil {
 		colorRed("\nError while creating the database %q: %s\n", dbPath, err)
 		os.Exit(1)
 	}
 
-	_, err = db.Exec(tableSchema)
+	err = sqlitex.Execute(conn, tableSchema, &sqlitex.ExecOptions{})
 	if err != nil {
+		// 	// TODO: add better err messege
```
Comment (pouriyajamshidi) [inline db.go:114]: Do we need this TODO?

File: db.go [hunk 3]
```diff
@@ -177,26 +179,48 @@ func (s database) saveStats(stat stats) error {
...
-		neverSucceedProbe, neverFailedProbe,
-		lastSuccessfulProbe, lastUnsuccessfulProbe,
-		totalPackets, packetLoss,
-		stat.totalUptime.String(), stat.totalDowntime.String(),
-		longestUptimeDuration, longestUptimeStart, longestUptimeEnd,
-		longestDowntimeDuration, longestDowntimeStart, longestDowntimeEnd,
-		fmt.Sprintf("%.3f", stat.rttResults.min), fmt.Sprintf("%.3f", stat.rttResults.average), fmt.Sprintf("%.3f", stat.rttResults.max),
-		stat.startTime.Format(timeFormat), stat.endTime.Format(timeFormat), totalDuration,
-	)
+	// datas
```
Comment (pouriyajamshidi) [inline db.go:182]:
```suggestion
	// data
```

**Review by pouriyajamshidi — APPROVED** [review]
Fantastic job @wizsk 

This is much cleaner than before

Thanks

File: db.go (hunk 2 above)
Comment (wizsk) [reply db.go:114]: if you think the error message is not stable then please subject a better error message.

Comment (pouriyajamshidi) [reply db.go:114]: I think it is fine for now

**Review by pouriyajamshidi — APPROVED** [review]

**Pull request merged**